sympy/core/sympify.py

```diff
@@ -416,7 +416,7 @@ def sympify(a, locals=None, convert_xor=True, strict=False, rational=False,
         flat = getattr(a, "flat", None)
```

Comment:
I don't like that this works by duck typing. The scalar check above doesn't. If it's going to do that, it should at least work using methods that are commonly available in other array libraries, which `flat` is not.

Reply:
Ideally we would make use of the [array API](https://data-apis.org/array-api/latest/), but that's not perfect because it isn't implemented everywhere yet (even NumPy itself only implements it as a sub-module).